iothub/device/src/ClientFactory.cs

```diff
@@ -80,27 +81,16 @@ public static InternalClient Create(string hostname, string gatewayHostname, IAu
 
             IotHubConnectionStringBuilder connectionStringBuilder = IotHubConnectionStringBuilder.Create(hostname, gatewayHostname, authenticationMethod);
 
+            // If the file upload transport settings hasn't been specified, we will create one using the client certificate on the connection string
```

Comment:
The client cert isn't a part of the connection string, it is supplied via the client's authentication mechanism provider.